src/components/Semester/Semester.vue

```diff
@@ -226,10 +235,11 @@ export default defineComponent({
 
   computed: {
     coursesForDraggable: {
-      get(): readonly FirestoreSemesterCourse[] {
+      get(): readonly (FirestoreSemesterCourse | FirestoreSemesterPlaceholder)[] {
         return this.courses;
       },
       set(newCourses: readonly AppFirestoreSemesterCourseWithRequirementID[]) {
+        // TODO: does this need to be updated? probably
         const courses = newCourses.map(({ requirementID: _, ...rest }) => rest);
```

Comment:
@benjamin-shen Looks like you wrote this code. What's the reasoning for leaving out `requirementID`?

Reply:
I think I wrote it. The courses with `requirementID` is dragged from requirements bar. It's used to indicate which requirement the user wants the course to match to.

Reply:
Sorry I didn't mean to leave this comment in there! I think this should be fine for now.